bpf/lib/conntrack.h

```diff
@@ -602,6 +703,11 @@ static __always_inline int ct_lookup4(const void *map,
 	else
 		return DROP_CT_INVALID_HDR;
 
+	/*
+	err = ct_extract_ports4(ctx, off, dir,
+				tuple, &action, &tcp_flags, &has_l4_header);
+	*/
```

Comment:
An argument for keeping it is that it explains why the various arguments that are not used in the current caller (`action`, `tcp_flags`, and `has_l4_header`) are there.

Reply:
:+1: to add a comment that explains that :slightly_smiling_face:

Reply:
> This should get inlined, no?

Yep, it was inlined.

> Is there any difference between the function above and the code below? That is, do you know why it might result in a complexity issue?

If think the error checking could have made a difference. However, it was not always reproducible :confused: 

> If we're not going ahead with ct_extract_ports4, I think we should remove it rather than keep dead code in the hope we'll be able to uncomment some day.

Are you referring to the commented out `ct_extract_port4` function invocation or the `ct_extract_port4` function itself?

@kkourt @pchaigno I'm going to remove the commented block, and also remove unused params in the `ct_extract_port4` function. Do we all ACK on this?

Reply:
> I'm going to remove the commented block, and also remove unused params in the ct_extract_port4 function. Do we all ACK on this?

:+1:

Reply:
ACK!